tests/test_task.py

```diff
@@ -1,196 +1,462 @@
-    task = setup_mock_task()
+    task = get_mock_task()
 
     setup_mock_namelists_directory()
-    setup_fluxnet_directory_tree([task], root_dir=TMP_DIR)
+    setup_mock_run_directory(task)
     do_mock_checkout_and_build()
 
-    task.fetch_files(root_dir=TMP_DIR)
-
-    assert Path(TMP_DIR, internal.SITE_TASKS_DIR,
-                task.get_task_name(), internal.CABLE_NML).exists()
-    assert Path(TMP_DIR, internal.SITE_TASKS_DIR,
-                task.get_task_name(), internal.CABLE_VEGETATION_NML).exists()
-    assert Path(TMP_DIR, internal.SITE_TASKS_DIR,
-                task.get_task_name(), internal.CABLE_SOIL_NML).exists()
-    assert Path(TMP_DIR, internal.SITE_TASKS_DIR,
-                task.get_task_name(), internal.CABLE_EXE).exists()
+    task.fetch_files()
+
+    assert Path(
+        MOCK_CWD, internal.SITE_TASKS_DIR, task.get_task_name(), internal.CABLE_NML
+    ).exists()
+    assert Path(
+        MOCK_CWD,
+        internal.SITE_TASKS_DIR,
+        task.get_task_name(),
+        internal.CABLE_VEGETATION_NML,
+    ).exists()
+    assert Path(
+        MOCK_CWD, internal.SITE_TASKS_DIR, task.get_task_name(), internal.CABLE_SOIL_NML
+    ).exists()
+    assert Path(
+        MOCK_CWD, internal.SITE_TASKS_DIR, task.get_task_name(), internal.CABLE_EXE
+    ).exists()
 
 
 def test_clean_task():
     """Tests for `clean_task()`."""
 
     # Success case: fetch then clean files
-    task = setup_mock_task()
+    task = get_mock_task()
 
     setup_mock_namelists_directory()
-    setup_fluxnet_directory_tree([task], root_dir=TMP_DIR)
+    setup_mock_run_directory(task)
     do_mock_checkout_and_build()
 
-    task.fetch_files(root_dir=TMP_DIR)
+    task.fetch_files()
 
     do_mock_run(task)
 
-    task.clean_task(root_dir=TMP_DIR)
+    task.clean_task()
+
+    assert not Path(
+        MOCK_CWD, internal.SITE_TASKS_DIR, task.get_task_name(), internal.CABLE_NML
+    ).exists()
+    assert not Path(
+        MOCK_CWD,
+        internal.SITE_TASKS_DIR,
+        task.get_task_name(),
+        internal.CABLE_VEGETATION_NML,
+    ).exists()
+    assert not Path(
+        MOCK_CWD, internal.SITE_TASKS_DIR, task.get_task_name(), internal.CABLE_SOIL_NML
+    ).exists()
+    assert not Path(
+        MOCK_CWD, internal.SITE_TASKS_DIR, task.get_task_name(), internal.CABLE_EXE
+    ).exists()
+    assert not Path(
+        MOCK_CWD, internal.SITE_OUTPUT_DIR, task.get_output_filename()
+    ).exists()
+    assert not Path(MOCK_CWD, internal.SITE_LOG_DIR, task.get_log_filename()).exists()
+
+
+def test_patch_namelist():
+    """Tests for `patch_namelist()`."""
+
+    nml_path = MOCK_CWD / "test.nml"
+
+    # Success case: patch non-existing namelist file
+    assert not nml_path.exists()
+    patch_namelist(nml_path, {"cable": {"file": "/path/to/file", "bar": 123}})
+    assert f90nml.read(nml_path) == {
+        "cable": {
+            "file": "/path/to/file",
+            "bar": 123,
+        }
+    }
 
```

Comment:
Can you put the patch dictionary in a variable and reuse that variable in the assert? Instead of rewriting the whole dictionary.

Reply:
Fixed in [95f1871](https://github.com/CABLE-LSM/benchcab/pull/99/commits/95f187105c25613915dc49b0185b4aebefff0f42)